core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -573,9 +576,12 @@ class CivilizationInfo {
         val cityToAddTo = city ?: cities.random()
         if (!gameInfo.ruleSet.units.containsKey(unitName)) return
         val unit = getEquivalentUnit(unitName)
-        placeUnitNearTile(cityToAddTo.location, unit.name)
-        if (unit.isGreatPerson())
-            addNotification("A [${unit.name}] has been born in [${cityToAddTo.name}]!", cityToAddTo.location, unit.name)
+        // silently bail if no tile to place the unit is found
+        val placedUnit = placeUnitNearTile(cityToAddTo.location, unit.name) ?: return
+        if (unit.isGreatPerson()) {
+            val locations = LocationAction(listOf(placedUnit.getTile().position,cityToAddTo.location))
```

Comment:
No need for the city location, I feel.

Reply:
OK but the null pointer fix stays - subject to out Elvis policy below?